### PR TITLE
Offline testing

### PR DESCRIPTION
**Pull request by jsumners**
This pull request re-works the whole test suite to use a mock server. Thus, the tests are able to be run without a hard dependency on any individual Active Directory server.

It also fixes a couple bugs that were shown as a result of the updates. Mainly that the default attributes were out-of-sync with what the documentation said they were. To alleviate this, I expose the `defaultAttributes` as a property on the `ActiveDirectory` function.

Further, I re-named the test suites to match their script names. This means the following commands are available for individual testing:
- `mocha --ui bdd --grep ^Attributes\s`
- `mocha --ui bdd --grep ^Authentication\s`
- `mocha --ui bdd --grep ^Network\s`
- `mocha --ui bdd --grep ^ctor\s`
- `mocha --ui bdd --grep ^find\s`
- `mocha --ui bdd --grep ^findGroup\s`
- `mocha --ui bdd --grep ^findGroups\s`
- `mocha --ui bdd --grep ^findUser\s`
- `mocha --ui bdd --grep ^findUsers\s`
- `mocha --ui bdd --grep ^getGroupMembershipForGroup\s`
- `mocha --ui bdd --grep ^getGroupMembershipForUser\s`
- `mocha --ui bdd --grep ^getUsersForGroup\s`
- `mocha --ui bdd --grep ^groupExists\s`
- `mocha --ui bdd --grep ^isUserMemberOf\s`
- `mocha --ui bdd --grep ^userExists\s`

I also switched the assertion library to Chai and updated all of the test code to ES2015 syntax. So the `assertmore` script got removed and the the underscore library is no longer used (in the tests).

### CODE REVIEW

**Comment by gheeres** [comment]
Thanks for the pull request. 

Please give me some time to review the changes. A quick glance looks **GREAT**!

**Comment by jsumners** [comment]
No problem. It's a lot to review.

File: test/mockServer/server.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+'use strict';
```
Comment (jsumners) [inline test/mockServer/server.js]: This file was used so that I could issue queries against it with ldapsearch. It should probably be removed.

**Comment by jsumners** [comment]
I have updated my branch with some clarification comments and elimination of merge conflicts.

**Pull request closed without merging**